allennlp/data/dataset_readers/wikitables/wikitables_dataset_reader.py

```diff
@@ -0,0 +1,141 @@
+
+    @overrides
+    def read(self, file_path):
+        instances = []
+        with open(file_path, "r") as data_file:
+            logger.info("Reading instances from lines in file: %s", file_path)
+            for line in tqdm.tqdm(data_file):
+                line = line.strip("\n")
+                if not line:
+                    continue
+                instances.append(self.text_to_instance(line, True))
+        if not instances:
+            raise ConfigurationError("No instances read!")
+        return Dataset(instances)
+
+    @overrides
+    def text_to_instance(self,  # type: ignore
+                         input_lisp_string: str,
+                         for_train: bool = False) -> Instance:
+        # pylint: disable=arguments-differ
+        parsed_info = self._parse_line_as_lisp(input_lisp_string)
+        tokenized_utterance = self._utterance_tokenizer.tokenize(parsed_info["utterance"])
+        utterance_field = TextField(tokenized_utterance, self._utterance_token_indexers)
+        table_filename = "%s/%s" % (self._tables_directory, parsed_info["context"])
+        # We want the TSV file, but the ``*.examples`` files typically point to CSV.
+        table_filename = table_filename.replace(".csv", ".tsv")
+        table_knowledge_graph = TableKnowledgeGraph.read_from_file(table_filename)
+        table_field = KnowledgeGraphField(table_knowledge_graph, self._utterance_token_indexers)
+        if for_train:
+            world = World(table_knowledge_graph)
+            dpd_output_filename = "%s/%s.gz" % (self._dpd_output_directory, parsed_info["id"])
+            sempre_forms = [line.strip().decode('utf-8') for line in gzip.open(dpd_output_filename)]
+            expressions = world.process_sempre_forms(sempre_forms)
+            action_sequences = [world.get_action_sequence(expression) for expression in expressions]
+            action_sequences_field = ListField([self._make_action_sequence_field(sequence)
+                                                for sequence in action_sequences])
+            return Instance({"utterance": utterance_field,
+                             "context": table_field,
+                             "action_sequences": action_sequences_field})
+        else:
+            return Instance({"utterance": utterance_field, "context": table_field})
+
+    @staticmethod
+    def _make_action_sequence_field(action_sequence: List[str]) -> ListField:
+        action_sequence.insert(0, START_SYMBOL)
+        action_sequence.append(END_SYMBOL)
+        return ListField([LabelField(action, label_namespace='actions') for action in action_sequence])
+
+    @staticmethod
+    def _parse_line_as_lisp(lisp_string: str) -> Dict[str, LispValueType]:
+        parsed_info = {}
+        input_nested_list = pyparsing.OneOrMore(pyparsing.nestedExpr()).parseString(lisp_string).asList()[0]
+        for key_value in input_nested_list:
```

Comment:
Can you do `for key, value in input_nested_list`?

Reply:
Done.